### PR TITLE
fix return type

### PR DESCRIPTION
**Pull request by thinkingmedia**
Adds type hinting for return.

### CODE REVIEW

File: src/Datasource/ModelAwareTrait.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ protected function _setModelClass($name)
      *
      * @param string|null $modelClass Name of model class to load. Defaults to $this->modelClass
      * @param string|null $modelType The type of repository to load. Defaults to the modelType() value.
-     * @return object The model instance created.
+     * @return \Cake\ORM\Table The model instance created.
```
Comment (dereuromark) [inline src/Datasource/ModelAwareTrait.php]: Shouldnt that be the interface instead?

Comment (thinkingmedia) [inline src/Datasource/ModelAwareTrait.php]: @dereuromark I looked for an interface, but didn't see one for Table. Which one should it be?

Comment (TheFRedFox) [inline src/Datasource/ModelAwareTrait.php]: I guess he means the `RepositoryInterface`. :wink:

Comment (Marlinc) [inline src/Datasource/ModelAwareTrait.php]: Yes, please use `\Cake\Datasource\RepositoryInterface`.

**Comment by thinkingmedia** [comment]
done

**Comment by dereuromark** [comment]
Thank you! :)

**Pull request merged**